subprojects/core/src/main/java/org/gradle/internal/build/DefaultProjectToolingModelBuilderLookupPreparer.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.build;
+
+import org.gradle.api.internal.project.ProjectState;
+import org.gradle.internal.buildtree.BuildModelParameters;
+
+public class DefaultProjectToolingModelBuilderLookupPreparer implements ProjectToolingModelBuilderLookupPreparer {
+
+    private final BuildModelParameters buildModelParameters;
+    private final BuildLifecycleController buildLifecycleController;
+
+    public DefaultProjectToolingModelBuilderLookupPreparer(BuildModelParameters buildModelParameters, BuildLifecycleController buildLifecycleController) {
+        this.buildModelParameters = buildModelParameters;
+        this.buildLifecycleController = buildLifecycleController;
+    }
+
+    @Override
+    public void prepare(ProjectState projectState) {
+        if (buildModelParameters.isIsolatedProjects()) {
+            buildLifecycleController.configureProjects();
```

Comment:
💭  Calling this methods here is actually redundant, because it is already called unconditionally on that model-building code path here. 

First via this call:
https://github.com/gradle/gradle/blob/4e67a794e52d87f7b304a241360286da438276e8/subprojects/core/src/main/java/org/gradle/internal/build/DefaultBuildToolingModelController.java#L61

And then via here:
https://github.com/gradle/gradle/blob/4e67a794e52d87f7b304a241360286da438276e8/subprojects/core/src/main/java/org/gradle/internal/build/DefaultBuildToolingModelController.java#L70-L71